src/Hangfire.Mongo/MongoJobFetcher.cs

```diff
@@ -115,12 +116,35 @@ private MongoFetchedJob TryGetEnqueuedJob(string queue, CancellationToken cancel
             });
             
             var update = new BsonDocument("$set", new BsonDocument(nameof(JobQueueDto.FetchedAt), DateTime.UtcNow));
-            
-            var fetchedJob = _dbContext
+
+            JobQueueDto fetchedJob;
+            try
+            {
+                fetchedJob = _dbContext
                 .JobGraph
                 .OfType<JobQueueDto>()
                 .FindOneAndUpdate(filter, update, Options, cancellationToken);
-            
+            }
+            catch (MongoCommandException ex)
+            {
+                var delayMs = 5000;
+                var regex = new Regex(@"RetryAfterMs=(\d+),");
+                var match = regex.Match(ex.Message);
+                if (match.Success)
+                {
+                    if (!int.TryParse(match.Groups[1].Value, out delayMs))
+                    {
+                        delayMs = 5000;
+                    }
+                }
+
+                Thread.Sleep(delayMs + 100);
+                fetchedJob = _dbContext
+                .JobGraph
+                .OfType<JobQueueDto>()
+                .FindOneAndUpdate(filter, update, Options, cancellationToken);
+            }
+
```

Comment:
does this benefit the mongodb implementation? or is this cosmodb specific? please inherit MongoJobFetcher and do the cosmosdb specific part in the class.
eg. CosmosDbJobFetcher : MongoJobFetcher

Reply:
Both, it would be useful for a remote Mongo Database